extraction/content_extractors/embedded_youtube_video_extractor.py

```diff
@@ -16,11 +18,11 @@ class EYouTubeVideoExtractor(IContentExtractor):
     def validate_and_extract(self, node: bs4.element):
         if node.name == 'iframe' and node.has_attr('src')\
                 and utils.has_domain(node['src'],
-                                     r'^https://www\.youtube\.com/embed'):
+                                     r'^https://www\.youtube\.com/embed/'):
             return EYouTubeVideo(self.__get_youtube_video_id(node['src']))
 
         return None
 
     @staticmethod
     def __get_youtube_video_id(url):
-        return url.split('/')[4].split('?')[0]
+        return urlparse(url)[2].split('/')[2]
```

Comment:
Do we need to split now?

Reply:
Is there any other way using `urlparse`?

Reply:
**urlparse** returns the **path** as a separate attribute which is at 2nd index (getting that using `urlparse(url)[2]`). Now we want the id from that path that's why using split to get that.

Reply:
Suppose the url is `https://www.youtube.com/embed/tgbNymZ7vqY` we will get `'/embed/tgbNymZ7vqY'` as path then using split getting **tgbNymZ7vqY**